pkg/controller/podautoscaler/horizontal_test.go

```diff
@@ -272,43 +273,84 @@ func (tc *testCase) prepareTestClient(t *testing.T) (*fake.Clientset, *metricsfa
 				podPhase = tc.reportedPodPhase[i]
 			}
 
+			podDeletionTimestamp := false
+			if tc.reportedPodDeletionTimestamp != nil {
+				podDeletionTimestamp = tc.reportedPodDeletionTimestamp[i]
+			}
+
 			podName := fmt.Sprintf("%s-%d", podNamePrefix, i)
 
 			reportedCPURequest := resource.MustParse("1.0")
 			if specifiedCPURequests {
 				reportedCPURequest = tc.reportedCPURequests[i]
 			}
 
-			pod := v1.Pod{
-				Status: v1.PodStatus{
-					Phase: podPhase,
-					Conditions: []v1.PodCondition{
-						{
-							Type:   v1.PodReady,
-							Status: podReadiness,
+			pod := v1.Pod{}
+			if podDeletionTimestamp == false {
+				pod = v1.Pod{
+					Status: v1.PodStatus{
+						Phase: podPhase,
+						Conditions: []v1.PodCondition{
+							{
+								Type:   v1.PodReady,
+								Status: podReadiness,
+							},
 						},
+						StartTime: &podStartTime,
 					},
-					StartTime: &podStartTime,
-				},
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      podName,
-					Namespace: namespace,
-					Labels: map[string]string{
-						"name": podNamePrefix,
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      podName,
+						Namespace: namespace,
+						Labels: map[string]string{
+							"name": podNamePrefix,
+						},
 					},
-				},
 
-				Spec: v1.PodSpec{
-					Containers: []v1.Container{
-						{
-							Resources: v1.ResourceRequirements{
-								Requests: v1.ResourceList{
-									v1.ResourceCPU: reportedCPURequest,
+					Spec: v1.PodSpec{
+						Containers: []v1.Container{
+							{
+								Resources: v1.ResourceRequirements{
+									Requests: v1.ResourceList{
+										v1.ResourceCPU: reportedCPURequest,
+									},
 								},
 							},
 						},
 					},
-				},
+				}
+			} else {
```

Comment:
you shouldn't be creating the entire object in the `if-else`, just setting the `DeletionTimestamp`:

```go
pod = v1.Pod{...}
if deletionTimestamp {
  pod.DeletionTimestamp = &metav1.Time{Time: time.Now()}
}
```